src/chunkserver/datastore/chunkserver_chunkfile.cpp

```diff
@@ -74,74 +75,131 @@ ChunkFileMetaPage& ChunkFileMetaPage::operator =(
     return *this;
 }
 
-void ChunkFileMetaPage::encode(char* buf) {
+void ChunkFileMetaPage::encode(char *buf) {
     size_t len = 0;
+
+    // uint8_t version 1 byte
     memcpy(buf, &version, sizeof(version));
     len += sizeof(version);
-    memcpy(buf + len, &sn, sizeof(sn));
-    len += sizeof(sn);
-    memcpy(buf + len, &correctedSn, sizeof(correctedSn));
-    len += sizeof(correctedSn);
+
+    // uint64_t sn 8 bytes need convert to big endian in encode
+    uint64_t beSn = htobe64(sn);
+    memcpy(buf + len, &beSn, sizeof(beSn));
+    len += sizeof(beSn);
+
+    // uint64_t correctedSn 8 bytes need convert to big endian
+    uint64_t beCorrectedSn = htobe64(correctedSn);
+    memcpy(buf + len, &beCorrectedSn, sizeof(beCorrectedSn));
+    len += sizeof(beCorrectedSn);
+
+    // long unsigned int loc_size 8 bytes need convert to big endian
     size_t loc_size = location.size();
-    memcpy(buf + len, &loc_size, sizeof(loc_size));
-    len += sizeof(loc_size);
+    size_t be_loc_size;
+    be_loc_size = htobe64(loc_size);
+    memcpy(buf + len, &be_loc_size, sizeof(be_loc_size));
+    len += sizeof(be_loc_size);
+
     // CloneChunk need serialized location information and bitmap information
     if (loc_size > 0) {
         memcpy(buf + len, location.c_str(), loc_size);
         len += loc_size;
+
+        // uint32_t bits 4 bytes need convert to big endian
         uint32_t bits = bitmap->Size();
-        memcpy(buf + len, &bits, sizeof(bits));
-        len += sizeof(bits);
+        uint32_t beBits = htobe32(bits);
+        memcpy(buf + len, &beBits, sizeof(beBits));
+        len += sizeof(beBits);
+
+        // unsigned long bitmapBytes 4 bytes
+        // bitmap char  (bits + 8 - 1) / 8 bytes
         size_t bitmapBytes = (bits + 8 - 1) >> 3;
         memcpy(buf + len, bitmap->GetBitmap(), bitmapBytes);
         len += bitmapBytes;
     }
+
+    // uint32_t crc 4 bytes
     uint32_t crc = ::curve::common::CRC32(buf, len);
-    memcpy(buf + len, &crc, sizeof(crc));
+    uint32_t beCrc = crc;
```

Comment:
Why is crc not converted to big endian？

Reply:
Beacue the crc is not defined in ChunkFileMetaPage and SnapshotMetapage.